### PR TITLE
Cleanup and reorganize the asm samples repo

### PR DESCRIPTION
**Pull request by xtineskim**
Addresses https://github.com/GoogleCloudPlatform/anthos-service-mesh-samples/issues/75
This is a large overhaul to re-organize the ASM samples repo.

- [ ] region tags (will require CL's to address these)
- [x] tf ci
- [x] acm config sync (~will open a corresponding PR in ACM repo~ https://github.com/GoogleCloudPlatform/anthos-config-management-samples/pull/57)

### CODE REVIEW

**Comment by ckim328** [comment]
I will open corresponding CL's for changing the region tags listed by snippet bot check

**Comment by ckim328** [comment]
https://github.com/GoogleCloudPlatform/anthos-config-management-samples/pull/57 
Draft PR opened for changing the ACM side

File: quickstarts/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Anthos Service Mesh Quickstarts
+
+This folder contains the quickstart material for creating a GKE cluster with Google's Managed Anthos Service Mesh.
+The following are methods showcased:
+* Terraform
```
Comment (NimJay) [inline quickstarts/README.md:5]: **Question:** Other than `terraform-asm-mcp/`, do we expect the `quickstarts/` folder to contain more folders in the future? Would it be overkill to have a separate `quickstarts/` folder?

File: quickstarts/README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+To view the other methods for setting up Managed Anthos Service Mesh:
+* [Fleet API](https://cloud.google.com/service-mesh/docs/managed/auto-control-plane-with-fleet)
+* [GCP Console](https://cloud.google.com/service-mesh/docs/install-anthos-service-mesh-console)
```
Comment (NimJay) [inline quickstarts/README.md:9]: **Thought:** I remember our rationale (from a previous sync) for not placing quickstart-related sample code inside the `tutorial-samples`: quickstarts don't necessarily rely on `cloud.google.com` tutorials. But now I'm thinking, should we make sure there is always a `cloud.google.com` for each quickstart in this repo — this is in line with the DPE philosophy of letting Tech Writers own the instructions/words. In this case, we could stuff `terraform-asm-mcp` into `tutorial-samples`.

File: quickstarts/README.md (hunk 1 above)
Comment (ckim328) [reply quickstarts/README.md:5]: I was thinking about this. I think having this folder can also lend itself to showing how to do a multi cluster quickstart w/ Terraform, or fleet api quickstart (either single cluster/multicluster). 

This quickstart just deploys the cluster and provisions ASM, and is not tied down to any sample app - I would ideally like to keep sample app setup vs day 0 provisioning separate. I think potentially the name `quickstarts` might be misleading, open to suggestions!

File: quickstarts/README.md (hunk 2 above)
Comment (ckim328) [reply quickstarts/README.md:9]: same point to above, we can discuss it off line and see where we want to go with this :)

File: shared/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+```
+kubectl create namespace asm-ingress
+```
+1. Enable auto-injection in the `asm-ingress` namespcae
+```
+kubectl lable namespace asm-ingress istio-injection=enabled istio.io/rev-
+```
+1. Deploy your Gateway
+```
+```
```
Comment (NimJay) [inline shared/README.md:35]: **Reminder:** Add command for "Deploy your Gateway".

Comment (ckim328) [reply shared/README.md:35]: yess! still a WIP for this section :)

**Comment by xtineskim** [comment]
Closing to break up this PR into smaller ones!

**Pull request closed without merging**